showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -127,6 +151,93 @@ class ShowkaseProcessor @JvmOverloads constructor(
             }.flatten().mapNotNull { it }.toSet()
     }
 
+    private fun processCustomAnnotation(
+        roundEnvironment: XRoundEnv,
+        annotation: XTypeElement? = null
+    ): Set<ShowkaseMetadata.Component> {
+        val supportedTypes = mutableListOf<String>()
+        if (annotation != null) supportedTypes.add(annotation.qualifiedName)
+        supportedTypes.addAll(supportedCustomAnnotationTypes())
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedTypes.map { supportedType ->
+            val annotatedElements = roundEnvironment.getElementsAnnotatedWith(supportedType)
+            annotatedElements.map { annotatedElement ->
+                if (!showkaseValidator.checkElementIsAnnotationClass(annotatedElement)) {
+
+                    showkaseValidator.validateComponentElementOrSkip(
+                        element = annotatedElement,
+                        annotationName = supportedType,
+                    )
+                    components.addAll(
+                        getShowkaseMetadataFromCustomAnnotation(
+                            element = annotatedElement,
+                            showkaseValidator = showkaseValidator,
+                            supportedType.getCustomAnnotationSimpleName(),
+                        ).toSet()
+                    )
+                }
+            }
+        }
+        return components
+    }
+
+    private fun String.getCustomAnnotationSimpleName(): String {
+        return this.split(".").last()
+    }
+
+    private fun processCustomAnnotationFromClasspath(roundEnvironment: XRoundEnv): Set<ShowkaseMetadata.Component> {
+        // In this function we are checking generated classpath for MultiPreview codegen annotations.
+        // We also check the current module if there is any composables that are annotated with the qualified name
+        // from the annotation from classpath. We use the fields from the classpath annotation to build
+        // common data for the ShowkasMetadata.
+
+        // Supported annotations from classpath
+        val supportedCustomPreview = environment.getTypeElementsFromPackage(CODEGEN_PACKAGE_NAME)
+            .flatMap { it.getEnclosedElements() }.mapNotNull {
+                when (
+                    val annotation = it.getAnnotation(ShowkaseMultiPreviewCodegenMetadata::class)
```

Comment:
if we are manually passing the custom annotation, then we don't need to create a separate codepath or generate any files with this custom annotation. We should be able to use the existing setup that `@Preview` and `@ShowkaseComposable` annotations are using. 

This approach was only needed if we were trying to infer the custom annotation. Is that what you are still hoping to do? Was the limitation that this was only working for ksp and not for kapt? I think we settled on just favoring consistency and going with the "manually specificying custom annotations" approach. At that point, we don't need this and should reuse the codepath used by `@Preview` and/or `@ShowkaseComposable`

Reply:
It's also possible I've forgotten some of the details we discussed earlier and I might be incorrect. Could you remind me?

Reply:
Hmm, actually there's the part where after identifying a custom annotation, we need to somehow "remember" the previews that it's annotated with and using that in the metadata. Where are you doing that part?

Reply:
I think I also have forgotten some of the details that we discussed. I think we made the decision on if we should use the compiler argument approach on the claim that the `ShowkaseMultiPreviewCodegenMetadata` approach only worked with KSP. However, I think I tested that with repeatable preview annotations in KAPT. When testing it now, it works without the compiler args. IMO it would be a better approach for the end user if we did not go with the compiler arg approach. Wdyt?